src/Development/IDE/Core/Shake.hs

```diff
@@ -260,15 +250,13 @@ data IdeState = IdeState
 
 
 -- This is debugging code that generates a series of profiles, if the Boolean is true
-shakeRunDatabaseProfile :: Maybe FilePath -> ShakeDatabase -> [Action a] -> IO ([a], Maybe FilePath)
-shakeRunDatabaseProfile mbProfileDir shakeDb acts = do
-        (time, (res,_)) <- duration $ shakeRunDatabase shakeDb acts
-        proFile <- for mbProfileDir $ \dir -> do
+shakeDatabaseProfile :: Maybe FilePath -> ShakeDatabase -> IO (Maybe FilePath)
+shakeDatabaseProfile mbProfileDir shakeDb =
+        for mbProfileDir $ \dir -> do
                 count <- modifyVar profileCounter $ \x -> let !y = x+1 in return (y,y)
-                let file = "ide-" ++ profileStartTime ++ "-" ++ takeEnd 5 ("0000" ++ show count) ++ "-" ++ showDP 2 time <.> "html"
```

Comment:
I found it quite useful to find the slow database runs quickly by seeing their time in the filename. Typically there are 6 0.00s profiles and one which takes 5s.

Reply:
With the Shake session there is no longer a single run being measured, the profiles measure all what happened for the duration of the session. 

In the absence of that metric, I considered tagging the profiles by the *duration* of the session, but I figured out this would be misleading.